frontend/dockerfile/dockerfile_lint_test.go

```diff
@@ -817,6 +817,18 @@ LABEL key value
 FROM scratch
 ENV key=value
 LABEL key=value
+`)
+	checkLinterWarnings(t, sb, &lintTestParams{Dockerfile: dockerfile})
+
+	// No warnings if the lint happens in a stage that's not reachable.
+	dockerfile = []byte(`
+FROM scratch AS a
+
+FROM a AS b
+ENV key value
+LABEL key value
+
+FROM a AS c
 `)
 	checkLinterWarnings(t, sb, &lintTestParams{Dockerfile: dockerfile})
```

Comment:
I think this test will not pass after rebase to https://github.com/moby/buildkit/pull/5000 and should needs to be rewritten like the test in that PR

Reply:
Rebased and updated.